### ISSUE TITLE
Best approach to implement set operation

### ISSUE THREAD

**Issue by arun-prasat** [ClickHouse/ClickHouse#23496]
(you don't have to strictly follow this form)

**Describe the issue**
I like to implement a set operation in clickhouse, Set operation will be dynamic and it may change in the runtime. Please let me know the best approach to implement the set operation clickhouse.

EX :
(Subquery A UNION ALL subquery B) UNION (Subquery C INTERSECT Subquery D) EXCEPT (Subquery E)

**Comment by vdimir** [comment]
There are plans about this feature, but I'm not sure about current status.
https://github.com/ClickHouse/ClickHouse/issues/17623#issuecomment-809275068
https://github.com/ClickHouse/ClickHouse/issues/15065

**Comment by arun-prasat** [comment]
ok thanks, but do you have any suggestion to implement the same with current clickhouse functionality(best approach).

**Comment by dengc367** [comment]
Yeah, I need this feature too, do you have a plan to do this feature? @alexey-milovidov

**Comment by alexey-milovidov** [comment]
INTERSECT and EXCEPT operators will be implemented not earlier than summer this year.
Right now you can use IN and NOT IN operators for this purpose.
